### DISCUSSION TITLE
Need to ensure sufficient "minimal" start apps for music client...

### DISCUSSION THREAD

**Discussion by alan12345678** [i3/i3#5359] [Q&A]
I hope this is the place to ask :)

I've been running i3 on Arch for some while now with zero issues. Slightly home rolled setup, which I'll explain so that my new issue might make sense.

I don't use a login manager. Single user, machine is not shared (a desktop, FWIW). I type in username/pwd at which point

~/.xinitrc is read. Standard fare, onto which I tacked my own stuff

`xgamma -gamma 0.9
setxkbmap -model pc86 -layout gb -variant colemak_dh
export QT_QPA_PLATFORMTHEME="qt5ct"
export XDG_HOME=${HOME}
export XDG_CONFIG_HOME=${HOME}/.config
export XDG_CACHE_HOME=${HOME}/.cache
export XDG_DATA_HOME=${HOME}/.local/share
picom -b &
lxqt-policykit-agent  &
mpd &
~/.fehbg &
user-dirs-dirs-update &
exec i3`

My i3 config is standard, onto which I added a couple of apps to open on start, e.g. browser, file manager, and geany. So far so good, it's been running fine for a couple of years and is the most stable setup I've ever used.

Today I discover that my long preferred music client (Cantata) is no longer supported. Okay, so it works for the time being, but I looked around for replacements that can behave the same. I run mpd locally, and have that all set up for audiophile playback, so I wanted another graphical front end. I installed both CoverBrowser (mcg) and mpdevil.

This is where my issues began. Both of these apps have to completely rebuild the library and cover art every time I start them. This takes ages, plus there's lots of blank covers (using the placeholder image). Obviously I know that every single music file is correct.

I got a bunch of errors like this from mcg

`(mcg:8444): Gtk-CRITICAL **: 18:03:24.390: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2023-01-08 18:04:15,991 ERROR: Failed to load albumart
Traceback (most recent call last):
  File "/usr/share/mcg/mcg/librarypanel.py", line 492, in _set_albums
    pixbuf = Utils.load_thumbnail(cache, self._client, album, size)
  File "/usr/share/mcg/mcg/utils.py", line 38, in load_thumbnail
    albumart = client.get_albumart_now(album.get_id())
  File "/usr/share/mcg/mcg/client.py", line 317, in get_albumart_now
    (_, albumart) = future.result()
TypeError: cannot unpack non-iterable NoneType object
`

I noted from the build file for mcg that it lists things like "gsettings-schema" as dependency, which I'm guessing is something that would be automatically started by a DM like Gnome, or even XFCE?

Anyone got any suggestions or helpful hints?

**Comment by neg-serg** [comment]
This is not related to i3, it's about X11 session management.

There is several ways to get what you need, if I understand the problem correctly.

1) I am personally use systemd service. You can check my dotfiles.

2) The second way is to try i3-gnome. It's not about gnome shell or gnome-flashback, but just allows you to get gnome stuff like keyring to be connected with i3. So it's pretty minimal.

3) Try regolith. It's minimal enough and also good example of session management.

I hope this some of these fix your problem.

**Reply by alan12345678** [reply]
Thank you for a constructive answer. It turns out that of the two additional music apps I was having problems with, neither was built to read cover images! A simple assumption on my part (that album art would "naturally" be read) turned out to be wrong.

One of the devs was super helpful (as so many here are) and released several updates written to fix the issue :)

So I'm happy that my minimal config remains fit for purpose, and that others care enough about their code that they respond to their users :)
